### ISSUE TITLE
pg_cron causes error when backup and restore DB

### ISSUE THREAD

**Issue by shirahoo** [citusdata/pg_cron#113]
Hi

When creating a backup for a DB with pg_cron extension installed
'CREATE POLICY cron_job_policy ON cron.job USING ((username = (CURRENT_USER)::text));' 
seems to appear twice.
This results with an error when restoring the DB from the backup file:
pg_restore: creating POLICY "cron.job cron_job_policy"
pg_restore: while PROCESSING TOC:
pg_restore: from TOC entry 3915; 3256 306262 POLICY job cron_job_policy 
pg_restore: error: could not execute query: ERROR:  policy "cron_job_policy" for table "job" already exists
Command was: CREATE POLICY cron_job_policy ON cron.job USING ((username = (CURRENT_USER)::text));

Please advise
Thank you

**Comment by lanm2434** [comment]
Hi, I have the exact same issue. Using Postresql 11.9 on Red Hat Linux 8

**Comment by shirahoo** [comment]
> Hi, I have the exact same issue. Using Postresql 11.9 on Red Hat Linux 8 and using pg_cron 1.3:
> Installed:
> pg_cron_11-1.3.0-1.rhel8.x86_64

Hi, the workaround I do meanwhile is "DROP POLICY cron_job_policy ON cron.job;" before dump/backup...

**Comment by marcocitus** [comment]
This seems to be a bug in postgres / pg_dump.

> Hi, the workaround I do meanwhile is "DROP POLICY cron_job_policy ON cron.job;" before dump/backup...

Thanks! Worth noting that non-superusers can no longer see their jobs in cron.job after dropping the policy, but they will still be able to schedule jobs as normal (assuming they have usage access on the cron schema).

**Comment by lanm2434** [comment]
Is the policy required? My understanding is that the polict is first created at the create extension pg_cron and then the policy is created again because the schema was dumped.  Any other idea how to resolve this? Dropping the policy before dumping would mean recreating it after ...

**Comment by shirahoo** [comment]
> Is the policy required? My understanding is that the polict is first created at the create extension pg_cron and then the policy is created again because the schema was dumped. Any other idea how to resolve this? Dropping the policy before dumping would mean recreating it after ...

You are right, policy appears twice in the dump file. When restoring, it is re-created by the extension, so before **each** dump it should be re-dropped (it's part of our dumping script)

**Comment by james-johnston-thumbtack** [comment]
This seems to be a documented limitation of PostgreSQL: https://www.postgresql.org/docs/14/extend-extensions.html

> PostgreSQL does not currently support extension scripts issuing CREATE POLICY or SECURITY LABEL statements. These are expected to be set after the extension has been created. All RLS policies and security labels on extension objects will be included in dumps created by pg_dump.

I just e-mailed the PG bugs mailing list, since there seems to not be a prior report: https://www.postgresql.org/message-id/CAANoKdbO%3Dy0FT9fSzP24-kMf_A5y9xyujwJQk3%3DHgDynZYMAzQ%40mail.gmail.com
